### PR TITLE
fix: Resolve HTTP 500 by replacing preferences by functions.

### LINKED ISSUE #1432: HTTP 500
**Describe the bug**

On pressing go fab a second time in search fragment with time field having such a value that would fetch no events, HTTP 500 error is thrown.

**To Reproduce**

1. Go to search fragment.
2. Select time as some value for which there are no event. eg. location Delhi, time Today on debug server.
3. press fab at bottom once.
4. come back to search fragment.
5. press fab again.
6. you will get error fetching events.

**Expected behavior**

Empty search results screen should be shown with any error.

**Logs**

2019-03-26 15:18:51.793 13597-13597/com.eventyay.attendee E/SearchViewModel$loadEvents: Error fetching events
    retrofit2.adapter.rxjava2.HttpException: HTTP 500 INTERNAL SERVER ERROR
        at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:54)
        at retrofit2.adapter.rxjava2.BodyObservable$BodyObserver.onNext(BodyObservable.java:37)
        at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:44)
        at io.reactivex.Observable.subscribe(Observable.java:12268)
        at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:34)
        at io.reactivex.Observable.subscribe(Observable.java:12268)
        at io.reactivex.internal.operators.observable.ObservableSingleSingle.subscribeActual(ObservableSingleSingle.java:35)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleFlatMap.subscribeActual(SingleFlatMap.java:36)
        at io.reactivex.Single.subscribe(Single.java:3603)
        at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)

**Screenshots**

ezgif com-video-to-gif

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by angmas1**
Fixes #1432 

Changes: HTTP 500 was shown because incorrect or null arguments were getting used in query sent to server. This was happening because certain time related variables were saved in preferences, and thus, their older value was getting used. Resolved this by using functions which fetch correct values before querying.

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt [hunk 1]
```diff
@@ -139,6 +141,56 @@ object Utils {
...
+    }
+
+    fun getNextToNextMonth(): String {
+        val calendar = Calendar.getInstance()
+        val today = calendar.get(Calendar.DAY_OF_MONTH)
+        val offset = 30 - today
+        calendar.add(Calendar.DATE, offset)
+        calendar.add(Calendar.MONTH, 1)
+        return getSimpleFormattedDate(calendar.time)
+    }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Move to DateTimeUtils

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: There is no such class/file, do you want me to create one or should do you want me to move to EventUtils??

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: Create

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: okay working on it

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/utils/Utils.kt]: done

**Pull request merged**